tools/cpp/runfiles/runfiles_src.h

```diff
@@ -93,6 +98,51 @@ class Runfiles {
  public:
   virtual ~Runfiles() {}
 
+#if defined(__has_builtin)
+#if __has_builtin(__builtin_FILE)
+#define BAZEL_TOOLS_CPP_RUNFILES_GENERIC_HAS_BUILTIN_FILE
+#endif
+#endif
+#if defined(_MSC_VER)
+#if _MSC_VER >= 1926
+#define BAZEL_TOOLS_CPP_RUNFILES_MSVC_HAS_BUILTIN_FILE
+#endif
+#endif
+#if defined(BAZEL_TOOLS_CPP_RUNFILES_GENERIC_HAS_BUILTIN_FILE) || \
+    defined(BAZEL_TOOLS_CPP_RUNFILES_MSVC_HAS_BUILTIN_FILE)
+#define BAZEL_TOOLS_CPP_RUNFILES_HAS_BUILTIN_FILE
+#endif
+#undef BAZEL_TOOLS_CPP_RUNFILES_GENERIC_HAS_BUILTIN_FILE
+#undef BAZEL_TOOLS_CPP_RUNFILES_MSVC_HAS_BUILTIN_FILE
+
+#if defined(BAZEL_TOOLS_CPP_RUNFILES_HAS_BUILTIN_FILE)
+  // Returns the repository name.
+  //
+  // Use this from within `cc_test` rules.
+  //
+  // Returns nullptr on error. If `error` is provided, the method prints an
+  // error message into it.
+  //
+  // This method looks at the RUNFILES_MANIFEST_FILE and TEST_SRCDIR
+  // environment variables.
+  //
+  // If source_repository is not provided, it defaults to the main repository
+  // (also known as the workspace).
+  static std::string CurrentRepository(const std::string& file = __builtin_FILE());
+ private:
+  static std::string GetPrefixHint() {
```

Comment:
Could you add a comment here explaining the purpose of this function? Doesn't have to be user-facing as it is private, but dev-facing.